### ISSUE TITLE
Unify transaction query view service RPCs

### ISSUE THREAD

**Issue by hdevalence** [penumbra-zone/penumbra#2311]
**Is your feature request related to a problem? Please describe.**

Currently we have somewhat awkward and inconsistent RPCs in the view service for getting transactions / transaction views / transaction perspectives:

- [`TransactionByHash`](https://buf.build/penumbra-zone/penumbra/docs/main:penumbra.view.v1alpha1#penumbra.view.v1alpha1.ViewProtocolService.TransactionByHash)
- [`TransactionPerspective`](https://buf.build/penumbra-zone/penumbra/docs/main:penumbra.view.v1alpha1#penumbra.view.v1alpha1.ViewProtocolService.TransactionPerspective)
- Nothing that gives a pre-built `TransactionView`
- [`Transactions`](https://buf.build/penumbra-zone/penumbra/docs/main:penumbra.view.v1alpha1#penumbra.view.v1alpha1.ViewProtocolService.Transactions) which streams a collection of `TransactionsResponse` (an entirely different object than a `TransactionByHashResponse`!)
- [`TransactionHashes`](https://buf.build/penumbra-zone/penumbra/docs/main:penumbra.view.v1alpha1#penumbra.view.v1alpha1.ViewProtocolService.TransactionHashes) which is like `Transactions` but only gives the hashes (it's not clear that we really save much bandwidth here)

**Describe the solution you'd like**

Unify and commonize these RPCs, handing out all relevant data and allowing the client to decide what they want to use. 

First, define a common
```proto
message TransactionInfo {
    // The height the transaction was included in a block, if known.
    uint64 height = 1;
    // The hash of the transaction.
    transaction.Id id = 2;
    // The transaction data itself.
    Transaction transaction = 3;
    // The transaction perspective, as seen by this view server.
    TransactionPerspective perspective = 4;
    // A precomputed transaction view of `transaction` from `perspective`, included for convenience of clients that don't have support for viewing transactions on their own.
    TransactionView view = 5;
}
```

Then, replace all of these RPC methods with just two:

- `TransactionByHash`, which returns a `TransactionInfo`;
- `Transactions`, which streams `TransactionInfo`s.

In other words, we change the view server to always return all possible information about the requested transactions.

**Comment by hdevalence** [comment]
It looks like the `TransactionPerspective` RPC now includes the full set of data, but the data isn't yet commonized / RPCs aren't deduplicated https://buf.build/penumbra-zone/penumbra/docs/main:penumbra.view.v1alpha1#penumbra.view.v1alpha1.ViewProtocolService.TransactionPerspective

**Comment by aubrika** [comment]
what do we think about naming the two remaining RPC methods `TransactionInfo` and 'TransactionInfoStream`?

**Comment by hdevalence** [comment]
`TransactionInfoByHash`/`TransactionInfoStream` sounds good to me!
